index.js

```diff
@@ -7,90 +7,74 @@ const defaultOptions = {
   apiKey: '' // the micro-mail api-key
 };
 
-exports.register = function(server, options, next) {
+const register = (server, options) => {
   options = Object.assign({}, defaultOptions, options);
   if (options.host === '' || options.apiKey === '') {
-    return next('You must specify a micro-mail host and apiKey!');
+    throw new Error('You must specify a micro-mail host and apiKey!');
   }
 
   const sendAddress = `${options.host}/send?token=${options.apiKey}`;
-  server.decorate('server', 'sendEmail', (data, done) => {
+  server.decorate('server', 'sendEmail', async(data) => {
     if (options.verbose) {
       server.log(['debug', 'hapi-micro-mail', 'send'], { message: `sending request to ${sendAddress}`, data });
     }
-    wreck.post(sendAddress, { payload: data }, (err, response, payload) => {
-      if (err) {
-        server.log(['error', 'hapi-micro-mail'], err);
-        if (typeof done === 'function') {
-          return done(err);
-        }
-        return;
-      }
-      payload = JSON.parse(payload.toString());
-      if (options.verbose) {
-        server.log(['debug', 'hapi-micro-mail'], { message: 'micro-mail server responded', data: payload });
-      }
-      if (typeof done === 'function') {
-        // handle all HTTP-level errors:
-        if (response.statusCode !== 200) {
-          return done(Boom.create(response.statusCode, response.statusMessage, payload));
-        }
-        // handle SMTP-level errors (return codes that are not '250')
-        // if there were multiple submissions sent with sendIndividual:
-        if (Array.isArray(payload.result)) {
-          // see if any were not successful:
-          let status = 'ok';
-          payload.result.forEach((result) => {
-            if (result.response.indexOf('250') < 0) {
-              status = 'error';
-            }
-          });
-          if (status !== 'ok') {
-            return done({
-              status,
-              message: 'failed to send one or more destination emails',
-              result: payload.result // will contain list of accepted/rejected emails
-            });
-          }
-        }
-        // if only one submission or sendIndividual was not true:
-        if (payload.result.response.indexOf('250') === -1) {
-          return done({
-            status: 'error',
-            message: payload.message,
-            result: payload.result
-          });
+    const { res, payload } = await wreck.post(sendAddress, { payload: data, json: true });
+    // payload = JSON.parse(payload.toString());
+    if (options.verbose) {
+      server.log(['debug', 'hapi-micro-mail'], { message: 'micro-mail server responded', data: payload });
+    }
+    // handle all HTTP-level errors:
+    if (res.statusCode !== 200) {
+      throw Boom.create(res.statusCode, res.statusMessage, payload);
```

Comment:
New version of boom doesn't have .create, we should just update boom and use the new form: `new Boom(res.statusMessage, { statusCode: res.statusCode, data: payload })`